e2e/framework/test_context.go

```diff
@@ -162,60 +165,63 @@ type TestContextType struct {
-	ForceLocalEndpoint bool `json:"local_endpoint,omitempty" yaml:"local_endpoint,omitempty" env:"ROBO_LOCAL_ENDPOINT"`
+	ForceLocalEndpoint bool `json:"local_endpoint,omitempty" yaml:"local_endpoint,omitempty" `
 	// ReportDir defines location to store the results of the test
-	ReportDir string `json:"report_dir" yaml:"report_dir" env:"ROBO_REPORT_DIR"`
+	ReportDir string `json:"report_dir" yaml:"report_dir" `
 	// ClusterName defines the name to use for domain name or state directory
-	ClusterName string `json:"cluster_name" yaml:"cluster_name" env:"ROBO_CLUSTER_NAME"`
+	ClusterName string `json:"cluster_name" yaml:"cluster_name" `
 	// License specifies the application license
-	License string `json:"license" yaml:"license" env:"ROBO_APP_LICENSE"`
+	License string `json:"license" yaml:"license" `
 	// OpsCenterURL specifies the Ops Center to use for tests.
 	// OpsCenterURL is mandatory when running tests on an existing Ops Center.
 	// In wizard mode, this is automatically populated by the wizard (incl. Application, see below)
-	OpsCenterURL string `json:"ops_url" yaml:"ops_url" env:"ROBO_OPS_URL"`
+	OpsCenterURL string `json:"ops_url" yaml:"ops_url" `
 	// Application defines the application package to test.
 	// In wizard mode, this is automatically set by the wizard
-	Application LocatorRef `json:"application" yaml:"application" env:"ROBO_APP"`
+	Application LocatorRef `json:"application" yaml:"application"`
 	// Login defines the login details to access existing Ops Center.
 	// Mandatory only in non-wizard mode
 	Login Login `json:"login" yaml:"login"`
 	// ServiceLogin defines the login parameters for service access to the Ops Center
 	ServiceLogin ServiceLogin `json:"service_login" yaml:"service_login"`
 	// FlavorLabel specifies the installation flavor label to use for the test.
 	// This is application-specific, e.g. `3 nodes` or `medium`
-	FlavorLabel string `json:"flavor_label" yaml:"flavor_label" env:"ROBO_FLAVOR_LABEL"`
+	FlavorLabel string `json:"flavor_label" yaml:"flavor_label" `
 
 	// AWS defines the AWS-specific test configuration
-	AWS AWSConfig `json:"aws" yaml:"aws"`
+	AWS *infra.AWSConfig `json:"aws" yaml:"aws"`
+	// Azure defines Azure cloud specific parameters
+	Azure *infra.AzureConfig `yaml:"azure"`
+
 	// Onprem defines the test configuration for bare metal tests
 	Onprem OnpremConfig `json:"onprem" yaml:"onprem"`
 	// Bandwagon defines the test configuration for post-install setup in bandwagon
 	Bandwagon BandwagonConfig `json:"bandwagon" yaml:"bandwagon"`
 	// WebDriverURL specifies optional WebDriver URL to use
-	WebDriverURL string `json:"web_driver_url,omitempty" yaml:"web_driver_url,omitempty" env:"ROBO_WEB_DRIVER_URL"`
+	WebDriverURL string `json:"web_driver_url,omitempty" yaml:"web_driver_url,omitempty" `
 	// Extensions groups arbitrary test step configuration
 	Extensions Extensions `json:"extensions,omitempty" yaml:"extensions,omitempty"`
 }
 
 type BandwagonConfig struct {
```

Comment:
that's pre-existing code; many structs meaning is unclear to me, and especially everything inside this test_context class is subject to massive changes in the coming days;